mobile/status.go

```diff
@@ -755,23 +756,53 @@ func ImageServerTLSCert() string {
 	return cert
 }
 
+type GetPasswordStrengthRequest struct {
+	Password   string   `json:"password"`
+	UserInputs []string `json:"userInputs"`
+}
+
+type PasswordScoreResponse struct {
+	Score int `json:"score"`
+}
+
 // GetPasswordStrength uses zxcvbn module and generates a JSON containing information about the quality of the given password
 // (Entropy, CrackTime, CrackTimeDisplay, Score, MatchSequence and CalcTime).
 // userInputs argument can be whatever list of strings like user's personal info or site-specific vocabulary that zxcvbn will
 // make use to determine the result.
 // For more details on usage see https://github.com/status-im/zxcvbn-go
 func GetPasswordStrength(paramsJSON string) string {
-	var params struct {
-		Password   string   `json:"password"`
-		UserInputs []string `json:"userInputs"`
+	var requestParams GetPasswordStrengthRequest
+
+	err := json.Unmarshal([]byte(paramsJSON), &requestParams)
+	if err != nil {
+		return makeJSONResponse(err)
 	}
 
-	err := json.Unmarshal([]byte(paramsJSON), &params)
+	data, err := json.Marshal(zxcvbn.PasswordStrength(requestParams.Password, requestParams.UserInputs))
+	if err != nil {
+		return makeJSONResponse(fmt.Errorf("Error marshalling to json: %v", err))
+	}
+	return string(data)
```

Comment:
Should it not be makeJSONResponse?

same question for function GetPasswordStrengthScore

Reply:
nvm